src/wdio/services/compare/LocalScreenshotProcessor.js

```diff
@@ -0,0 +1,106 @@
+    const parentName = LocalScreenshotProcessor.createTestName(context.test.parent);
+    const testName = LocalScreenshotProcessor.createTestName(context.test.title);
+
+    return `${parentName}[${testName}].png`;
+  }
+
+  static getFormFactor(context) {
+    let formFactor = global.browser.options.formFactor;
+
+    if (!formFactor) {
+      const browserWidth = context.meta.viewport.width;
+      const viewportSizes = Object.keys(VIEWPORTS);
+      viewportSizes.some((viewport) => {
+        if (browserWidth <= VIEWPORTS[viewport].width) {
+          formFactor = VIEWPORTS[viewport].name;
+          return true;
+        }
+        return false;
+      });
+    }
+
+    // If viewport value is undefined due to being larger then the terra defiend viewports, set formFactor to enormous.
+    formFactor = formFactor || 'enormous';
+
+    return formFactor;
+  }
+
+  static getScreenshotDir(context) {
+    const locale = global.browser.options.locale || 'en';
+    const browserName = context.desiredCapabilities.browserName;
+    const formFactor = LocalScreenshotProcessor.getFormFactor(context);
+    const testForm = `${browserName}_${formFactor}`;
+    const testSuite = path.parse(context.test.file).name;
+
+    return path.join(locale, testForm, testSuite);
+  }
+
+  static getScreenshotPath(ref, context) {
+    let testPath = path.dirname(context.test.file);
+
+    const baseDir = global.browser.options.baseScreenshotDir;
+    if (baseDir) {
+      testPath = testPath.split(process.cwd())[1];
+      // Added to allow for test reusablility from terra repositories
+      if (testPath.includes('node_modules')) {
+        testPath = testPath.split('node_modules')[1];
+      }
+      testPath = baseDir + testPath;
+    }
+    const refDir = screenshotSetup[`${ref}Dir`];
+
+    return path.join(testPath, '__snapshots__', refDir, LocalScreenshotProcessor.getScreenshotDir(context), LocalScreenshotProcessor.getScreenshotName(context));
+  }
+
+  static async getReferenceScreenshotData(context) {
+    const referenceScreenshotPath = LocalScreenshotProcessor.getScreenshotPath('reference', context);
+    return fs.readFile(referenceScreenshotPath);
+  }
+
+  static async saveLatestScreenshotData(context, latestScreenshotData) {
+    const latestScreenshotPath = LocalScreenshotProcessor.getScreenshotPath('screenshot', context);
+    await fs.outputFile(latestScreenshotPath, latestScreenshotData);
+  }
+
+  static async saveDiffFileData(context, diffData) {
+    const diffScreenshotPath = LocalScreenshotProcessor.getScreenshotPath('diff', context);
+    await fs.outputFile(diffScreenshotPath, diffData);
+  }
+
+  static async handleNoReferenceScreenshot(context, latestScreenshotData, createResultReport) {
+    const referenceScreenshotPath = LocalScreenshotProcessor.getScreenshotPath('reference', context);
+    await fs.outputFile(referenceScreenshotPath, latestScreenshotData);
+    return createResultReport(0, true, true);
```

Comment:
Talked offline:

- eventually want to fail for new reference if in `process.env.CI = true`
- could add console messages for now & warn when running in CI for visibility

Reply:
Or we could configure it for now as opt in and change it to the default in an MVB